File: app/src/main/java/com/letianpai/robot/desktop/ui/view/BackButton.java
package com.letianpai.robot.desktop.ui.view;

import android.content.Context;
import android.util.AttributeSet;

import com.letianpai.robot.desktop.R;
import com.letianpai.robot.desktop.utils.KeyConsts;

public class BackButton extends KeyImageButton{
    public BackButton(Context context) {
        super(context);
        initData();
    }
    public BackButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        initData();
    }

    public BackButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initData();
    }

    public BackButton(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initData();
    }

    private void initData() {
        keyImageButton.setPressedImage(R.drawable.back_press);
        keyImageButton.setUnPressedImage(R.drawable.back_normal);
        keyImageButton.setImagePosition(KeyConsts.CENTER,0,0,0,0,40,40);
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/ui/view/BackButton.kt
package com.letianpai.robot.desktop.ui.view

import android.content.Context
import android.util.AttributeSet
import com.letianpai.robot.desktop.R
import com.letianpai.robot.desktop.utils.KeyConsts

class BackButton : KeyImageButton {
    constructor(context: Context?) : super(context) {
        initData()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        initData()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initData()
    }

    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes) {
        initData()
    }

    private fun initData() {
        keyImageButton!!.setPressedImage(R.drawable.back_press)
        keyImageButton!!.setUnPressedImage(R.drawable.back_normal)
        keyImageButton!!.setImagePosition(KeyConsts.Companion.CENTER, 0, 0, 0, 0, 40, 40)
    }
}
